src/components/v5/common/CompletedAction/partials/StreamingPayment/partials/CancelModal/CancelModal.tsx

```diff
@@ -0,0 +1,140 @@
+import useAsyncFunction from '~hooks/useAsyncFunction.ts';
+import { ActionTypes } from '~redux';
+import { type CancelStreamingPaymentPayload } from '~redux/types/actions/expenditures.ts';
+import { Form } from '~shared/Fields/index.ts';
+import { formatText } from '~utils/intl.ts';
+import DecisionMethodSelect from '~v5/common/CompletedAction/partials/PaymentBuilder/partials/DecisionMethodSelect/DecisionMethodSelect.tsx';
+import Button from '~v5/shared/Button/index.ts';
+import Modal from '~v5/shared/Modal/index.ts';
+
+import { cancelDecisionMethodItems, validationSchema } from './consts.ts';
+import { type CancelModalProps } from './types.ts';
+import { getCancelDecisionMethodDescriptions } from './utils.ts';
+
+const CancelModal: FC<CancelModalProps> = ({
+  isOpen,
+  onClose,
+  streamingPayment,
+  refetchStreamingPayment,
+  ...rest
+}) => {
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+
+  const cancel = useAsyncFunction({
+    submit: ActionTypes.STREAMING_PAYMENT_CANCEL,
+    error: ActionTypes.STREAMING_PAYMENT_CANCEL_ERROR,
+    success: ActionTypes.STREAMING_PAYMENT_CANCEL_SUCCESS,
+  });
+
+  const handleCancelStreamingPayment = async ({
+    shouldWaive,
+  }: Pick<CancelStreamingPaymentPayload, 'shouldWaive'>) => {
+    try {
+      if (!streamingPayment) {
+        return;
+      }
+
+      const payload: CancelStreamingPaymentPayload = {
+        colonyAddress: colony.colonyAddress,
+        streamingPayment,
+        userAddress: user?.walletAddress ?? '',
+        shouldWaive,
+      };
+
+      await cancel(payload);
+
+      await refetchStreamingPayment({
+        streamingPaymentId: streamingPayment.id,
+      });
+
+      onClose();
+    } catch (err) {
+      onClose();
+    }
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} icon={Prohibit} {...rest}>
+      <h5 className="mb-2 heading-5">
+        {formatText({
+          id: 'cancelModal.locked.title',
+        })}
+      </h5>
+      <p className="mb-6 text-md text-gray-600">
+        {formatText(
+          {
+            id: 'cancelModal.locked.description',
+          },
+          {
+            role: formatText({ id: 'role.mod' }),
+          },
+        )}
+      </p>
+      <Form
+        className="flex flex-grow flex-col"
+        onSubmit={() => handleCancelStreamingPayment({ shouldWaive: false })}
+        validationSchema={validationSchema}
+        defaultValues={{ decisionMethod: {} }}
+      >
+        {({ watch, formState: { isSubmitting } }) => {
+          const method = watch('decisionMethod');
+          const { roleToShow } = getMemberPermissions(
+            colony,
+            user?.walletAddress || '',
+            streamingPayment.nativeDomainId,
+          );
+
+          const cancelDecisionMethodDecscriptions =
+            getCancelDecisionMethodDescriptions(
+              roleToShow?.name || formatText({ id: 'role.mod' }),
```

Comment:
This seems like a lot of leg work to show the current user role, which is not necessarily the 'minimum' role that is required. Would it be easier to hardcode the role, even if it means having different i18n messages for various cancel modals?

Reply:
I've asked about it at the daily meeting and I have to show the current user role here